src/components/API.js

```diff
@@ -0,0 +1,115 @@
+import { apiOptions } from "../utils/constants.js";
+
+export default class Api {
+  constructor() {
+    // constructor body
+    this._baseUrl = apiOptions.baseUrl;
+    this._headers = apiOptions.defaultHeaders;
+    this._userRoute = `${this._baseUrl}/users/me`;
+    this._cardRoute = `${this._baseUrl}/cards`;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  async _request(url, options) {
+    return await fetch(url, options).then(this._checkResponse);
+  }
+
+  // User route methods
+
+  fetchUserInfo() {
+    return this._request(this._userRoute, {
+      headers: this._headers,
+    });
+  }
+
+  editUserInfo({ name, about }) {
+    return this._request(this._userRoute, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    }).catch((err) => {
```

Comment:
Needs correction: (fix this for every api method) Errors should be handled outside of the api class in `index.js`. Now `catch` block is placed in the middle of the chain, so the rest of the chain won't know anything about the error — but it should because we need to update the data only in case of success

The api call in `index.js` should look like this:

```js
// show loading
api.
  .then(...)
  .catch(...)
  .finally(() => {
    // hide loading
  })
```